src/components/ont/runs/ONTFlowcell.vue

```diff
@@ -121,15 +119,21 @@ export default {
         this.setBarcodeState(response.success)
       },
     },
-    ...mapState({
-      flowcellId(state) {
-        const flowcell = state.currentRun.flowcell_attributes.find(
+    flowcellId: {
+      get() {
+        const flowcell = this.currentRun().flowcell_attributes.find(
           (flowcell) => flowcell.position == this.position,
         )
         if (flowcell) {
           return flowcell.flowcell_id
         }
+        return ''
       },
+      set(value) {
+        this.setFlowcellId({ $event: value, position: this.position })
+      },
+    },
+    ...mapState({
       poolTubeBarcode(state) {
         const flowcell = state.currentRun.flowcell_attributes.find(
```

Comment:
should this also be` this.currentRun()`?

Reply:
Because this is in mapState it has access to state directly so doesn't need the getter.

Reply:
I understand. Just wondered why it was done differently here. I can see in other places we modify via the getter but here we are modifying state and using the getter. I suspect this is historical.